File: mutate4go/process_command_executor_test.go
package mutate4go

import "testing"

func TestProcessCommandExecutorCapturesSuccessfulCommandOutput(t *testing.T) {
	result, err := ProcessCommandExecutor{}.Run([]string{"sh", "-c", "printf 'ok'"}, t.TempDir(), 0)
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	if result.ExitCode != 0 || result.Output != "ok" || result.TimedOut {
		t.Fatalf("unexpected result: %#v", result)
	}
}

func TestProcessCommandExecutorReturnsTimeoutExitCodeWhenCommandTakesTooLong(t *testing.T) {
	result, err := ProcessCommandExecutor{}.Run([]string{"sh", "-c", "sleep 1"}, t.TempDir(), 10)
	if err != nil {
		t.Fatalf("unexpected error: %v", err)
	}
	if result.ExitCode != 124 || !result.TimedOut {
		t.Fatalf("unexpected result: %#v", result)
	}
}

File: mutate4rs/tests/process_command_executor_test.rs
use mutate4rs::ProcessCommandExecutor;

#[test]
fn captures_successful_command_output() {
    let result = ProcessCommandExecutor
        .run(&["sh", "-c", "printf ok"], &std::env::temp_dir(), 0)
        .unwrap();

    assert_eq!(0, result.exit_code);
    assert_eq!("ok", result.output);
    assert!(!result.timed_out);
}

#[test]
fn returns_timeout_exit_code_when_command_takes_too_long() {
    let result = ProcessCommandExecutor
        .run(&["sh", "-c", "sleep 1"], &std::env::temp_dir(), 10)
        .unwrap();

    assert_eq!(124, result.exit_code);
    assert!(result.timed_out);
}
